azurerm/resource_arm_application_insights_api_key_test.go

```diff
@@ -0,0 +1,282 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMApplicationInsightsAPIKey_no_permission(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMApplicationInsightsAPIKey_basic(ri, testLocation(), "[]", "[]")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationInsightsAPIKeyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      config,
+				ExpectError: regexp.MustCompile("The API Key needs to have a Role"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMApplicationInsightsAPIKey_requiresImport(t *testing.T) {
+	if !requireResourcesToBeImported {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_application_insights_api_key.test"
+	ri := acctest.RandInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationInsightsAPIKeyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApplicationInsightsAPIKey_basic(ri, testLocation(), "[]", `["annotations"]`),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsAPIKeyExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "read_permissions.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "write_permissions.#", "1"),
+				),
+			},
+			{
+				Config:      testAccAzureRMApplicationInsightsAPIKey_requiresImport(ri, location, "[]", `["annotations"]`),
```

Comment:
What are we testing here 🤔

Reply:
Not sure either :p, it's mostly replicating what's done for other resources of this provider. Never seen this sort of tests elsewhere.

Reply:
After further digging, this kind of tests are somewhat new: https://github.com/terraform-providers/terraform-provider-azurerm/commit/2ba4f4e8e73f149c087fc5b59a63749f872451f3#diff-0f12f6d1a183efd9a355593877b4abebR39

As the author of that change, maybe @tombuildsstuff can advise whether this one should be kept.

Reply:
Yep, upon further investigation they should should be 🙂